pkg/machines/components/vcpuModal.jsx

```diff
@@ -0,0 +1,239 @@
+        if (divs.indexOf(this.state.cores) === -1) {
+            state.cores = divs[divs.length - 1];
+        }
+
+        // According to: Max value = Sockets * Cores * Threads. Threads = Max value / ( Sockets * Cores )
+        state.threads = state.max / ( state.cores * state.sockets );
+        this.setState(state);
+    }
+
+    onCountSelect (e) {
+        let value = parseInt(e.target.value);
+        value = clamp(value, this.state.max, 1);
+        this.setState({ count: parseInt(value) });
+    }
+
+    onSocketChange (value) {
+        let state = { sockets: this.state.sockets, cores: this.state.cores };
+        state.sockets = parseInt(value);
+
+        // Get divisors of Max VCPU number divided by number of sockets
+        let divs = dividers(this.state.max / state.sockets);
+
+        // If current cores value is not in divisors array, then change it to max divisor
+        if (divs.indexOf(this.state.cores) === -1) {
+            state.cores = divs[divs.length - 1];
+        }
+
+        // Likewise: Max value = Sockets * Cores * Threads. Sockets = Max value / ( Threads * Cores )
+        state.threads = (this.state.max / (state.sockets * state.cores));
+        this.setState(state);
+    }
+
+    onThreadsChange (value) {
+        let state = { sockets: this.state.sockets, threads: this.state.threads };
+        state.threads = parseInt(value);
+        let divs = dividers(this.state.max / state.threads);
+
+        // If current sockets value is not in divisors array, then change it to max divisor
+        if (divs.indexOf(state.sockets) === -1) {
+            state.sockets = divs[divs.length - 1];
+        }
+
+        // Likewise: Max value = Sockets * Cores * Threads. Cores = Max value / ( Threads * Sockets )
+        state.cores = (this.state.max / (state.sockets * state.threads));
+
+        this.setState(state);
+    }
+
+    onCoresChange (value) {
+        let state = { sockets: this.state.sockets, threads: this.state.threads };
+        state.cores = parseInt(value);
+
+        let divs = dividers(this.state.max / state.cores);
+
+        // If current sockets value is not in divisors array, then change it to max divisor
+        if (divs.indexOf(state.sockets) === -1) {
+            state.sockets = divs[divs.length - 1];
+        }
+
+        // Likewise: Max value = Sockets * Cores * Threads. Threads = Max value / ( Cores * Sockets )
+        state.threads = (this.state.max / (state.sockets * state.cores));
+        this.setState(state);
+    }
+
+
+    render () {
+        let caution = null;
+
+        if (this.props.isRunning) {
+            caution = (
+                <tr>
+                    <td colSpan={2} className="machines-vcpu-caution">
+                        <div className="alert alert-warning">
+                            <span className="pficon pficon-warning-triangle-o"></span>
+                            All changes will take effect only <strong>after reboot</strong>.
```

Comment:
Didn't spot that in my previous round: This needs to be i18n'ed, i. e. wrapped into `_("")`. This won't survive HTML tags, but I don't think they are important here - the whole notification is already quite well emphasized.